scripts/missions/amk/helpers.lua

```diff
@@ -78,128 +70,273 @@ local digZoneIds =
+            { x = 294.342, z = 211.039 },   -- (J-7)
+            { x = 122.505, z = 432.139 },   -- (I-6)
+            { x = -122.441, z = 444.200 },  -- (H-6)
+            { x = -130.948, z = 203.126 },  -- (H-7)
+        },
     },
-    -- 11 = Western Altepa Desert
-    [11] =
+    [xi.zone.YHOATOR_JUNGLE] =
     {
-        { x = -37, z = 398 },
+        eventID = 10,
+        spots =
+        {
+            { x = -2.428, z = -359.037 },   -- (H-10)
+            { x = -397.683, z = -260.857 }, -- (F-9) Under the log
+            { x = -255.970, z = -123.890 }, -- (F-9) NE of tele
+            { x = -385.761, z = -383.388 }, -- (F-10)
+            { x = 186.28, z = -187.310 },   -- (I-9)
+            { x = -83.513, z = 224.256 },   -- (G-7) Directly under the signpost
+            { x = -231.969, z = -144.198 }, -- (G-9) Just east of telepoint
+        },
     },
-    -- 12 = Eastern Altepa Desert
-    [12] =
+    [xi.zone.WESTERN_ALTEPA_DESERT] =
     {
-        { x = 14, z = 187 },
+        eventID = 11,
+        spots =
+        {
+            { x = -811.000, z = -396.380 }, -- (D-10)
+            { x = -637.710, z = -676.310 }, -- (E-11)
+            { x = -429.9, z = -33.000 },    -- (F-7)
+            { x = -165.390, z = -172.490 }, -- (H-8)
+            { x = 90.920, z = 50.630 },     -- (I-7)
+            { x = 567.740, z = 230.710 },   -- (L-6)
+            { x = 571.580, z = -10.900 },   -- (L-7) SE corner between two large cacti
+            { x = 674.245, z = 46.084 },    -- (M-7)
+            { x = -3.805, z = 348.229 },    -- (I-5)
+            { x = 334.699, z = 105.761 },   -- (K-7) NW corner
+        },
+    },
+    [xi.zone.EASTERN_ALTEPA_DESERT] =
+    {
+        eventID = 12,
+        spots =
+        {
+            { x = -309.358, z = 49.217 },   -- (F-8) NW corner
+            { x = 419.165, z = 38.675 },    -- (J-8) NE of center
+            { x = 368.639, z = -208.379 },  -- (J-9) SW corner
+            { x = 37.012, z = -393.275 },   -- (H-10) SW corner
+            { x = -197.142, z = 285.642 },  -- (F-6) SE corner
+            { x = 20.214, z = 73.494 },     -- (H-8) NW corner
+            { x = -189.196, z = 32.195 },   -- (F-8) NE corner
+            { x = 85.507, z = 313.316 },    -- (H-6) Center
+            { x = 469.810, z = 222.405 },   -- (J/K-7) Upper third of square
+            { x = -215.823, z = -181.996 }, -- (F-9) Center
+        },
     },
 }
 
-xi.amk.helpers.tryRandomlyPlaceDiggingLocation = function(player)
-    local diggingZoneOffset = xi.amk.helpers.getDiggingZone(player)
-    local diggingSiteTable  = xi.amk.helpers.digSites[diggingZoneOffset]
+xi.amk.helpers.getDiggingZone = function(player)
+    -- Returns zone id of 1 of the 13 possible zones, and sets variable
+    local diggingZone = player:getCharVar('Mission[10][6]diggingZone')
 
-    player:setLocalVar('AMK_DIG_SITE_INDEX', math.random(1, #diggingSiteTable))
+    if diggingZone == 0 then
```

Comment:
Should probably just change to check if `xi.amk.helpers.digSites[diggingZone] == nil` This will catch zero but also bunk values.

Storing the actual zone instead of an arbitray index is A++ though from a GM perspective.